### PR TITLE
Charts 8453 add group btns for collection workbook page

### PR DESCRIPTION
**Pull request by Sergey-weber**

### CODE REVIEW

File: src/ui/units/workbooks/components/WorkbookPage/WorkbookPage.tsx [hunk 1]
```diff
@@ -38,6 +41,8 @@ import {WorkbookTabs} from '../WorkbookTabs/WorkbookTabs';
 import {TAB_ALL} from '../WorkbookTabs/constants';
 import {TabId} from '../WorkbookTabs/types';
 
+import PencilToLineIcon from '@gravity-ui/icons/svgs/pencil-to-line.svg';
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookPage/WorkbookPage.tsx:44]: Let's use import from the lib root – `import {PencilToLine} from '@gravity-ui/icons';`

File: src/ui/units/collections/components/CollectionPage/CollectionPage.tsx [hunk 2]
```diff
@@ -45,6 +46,8 @@ import {CollectionActionPanel} from '../CollectionActionPanel/CollectionActionPa
 import {CollectionActions} from '../CollectionActions/CollectionActions';
 import {CollectionLayout} from '../CollectionLayout/CollectionLayout';
 
+import PencilToLineIcon from '@gravity-ui/icons/svgs/pencil-to-line.svg';
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionPage/CollectionPage.tsx:49]: `import {PencilToLine} from '@gravity-ui/icons';`

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 3]
```diff
@@ -22,6 +22,7 @@ import Utils from '../../../../utils';
 import collectionIcon from '../../../../assets/icons/collections/collection.svg';
 import workbookDemoIcon from '../../../../assets/icons/collections/workbook-demo.svg';
 import workbookIcon from '../../../../assets/icons/collections/workbook.svg';
+import LockOpenIcon from '@gravity-ui/icons/svgs/lock-open.svg';
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:25]: `import {LockOpen} from '@gravity-ui/icons';`

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx [hunk 4]
```diff
@@ -1,30 +1,28 @@
...
 import {WorkbookWithPermissions} from '../../../../../shared/schema';
 import {IamAccessDialog} from '../../../../components/IamAccessDialog/IamAccessDialog';
 import {registry} from '../../../../registry';
 import {ResourceType} from '../../../../registry/units/common/types/components/IamAccessDialog';
-import {AppDispatch} from '../../../../store';
-import {closeDialog, openDialog} from '../../../../store/actions/dialog';
 import Utils from '../../../../utils';
 import {CreateEntry} from '../CreateEntry/CreateEntry';
 
+import LockOpenIcon from '@gravity-ui/icons/svgs/lock-open.svg';
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:19]: `import {LockOpen} from '@gravity-ui/icons';`

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 5]
```diff
@@ -166,6 +153,15 @@ export const CollectionActions = React.memo<Props>(
                         }
                     />
                 )}
+                {collectionData && (
+                    <React.Fragment>
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:157]: Unnecessary `<React.Fragment>`

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 6]
```diff
@@ -135,29 +135,16 @@ export const CollectionActions = React.memo<Props>(
...
-                                {i18n('action_move')}
-                            </Button>
-                        )}
-                        {collectionData.permissions.update && (
-                            <Button className={b('edit')} onClick={onEditClick}>
-                                {i18n('action_edit')}
-                            </Button>
-                        )}
-                    </React.Fragment>
+                {collectionData && collectionData.permissions.move && (
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:138]: Let's place the "Move" button in the same place as on the workbook page, in the `...` button.

image

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx [hunk 7]
```diff
@@ -62,117 +59,91 @@ export const WorkbookActions: React.FC<Props> = ({workbook, refreshWorkbookInfo}
...
+            {Boolean(dropdownActions.length) && (
+                <DropdownMenu
+                    defaultSwitcherProps={{view: 'normal'}}
+                    switcherWrapperClassName={b('item')}
+                    items={dropdownActions}
+                />
             )}
 
-            {workbook.permissions.copy && (
+            {collectionsAccessEnabled && workbook.permissions.listAccessBindings && (
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:135]: 1) Incorrect order, should be – "Create", "Access" and "...".
2) There are no icons, but they are in the design.

image

image

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 8]
```diff
@@ -135,29 +135,16 @@ export const CollectionActions = React.memo<Props>(
 
         return (
             <div className={b(null, className)}>
-                {collectionData && (
-                    <React.Fragment>
-                        {collectionsAccessEnabled && collectionData.permissions.listAccessBindings && (
-                            <Button className={b('access')} onClick={onEditAccessClick}>
-                                {i18n('action_access')}
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:142]: `action_access` and `action_edit` from i18n are no longer used, let's remove them

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 6 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:138]: if
Screenshot 2023-10-04 at 10 23 02
 i will place button "Move", in button ... will be only 1 item

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 8 above)
Comment (imsitnikov) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:142]: upd: no need to remove them, they will be used in tooltips

File: src/ui/units/workbooks/components/WorkbookPage/WorkbookPage.tsx (hunk 1 above)
Comment (Sergey-weber) [reply src/ui/units/workbooks/components/WorkbookPage/WorkbookPage.tsx]: fixed

File: src/ui/units/collections/components/CollectionPage/CollectionPage.tsx (hunk 2 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionPage/CollectionPage.tsx]: fixed

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 3 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx]: fixed

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx (hunk 4 above)
Comment (Sergey-weber) [reply src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx]: fixed

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 5 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx]: fixed

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 6 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:138]: added new btn move, near btn ...

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx (hunk 7 above)
Comment (Sergey-weber) [reply src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:160]: fixed

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 8 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:142]: added tooltips

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx (hunk 7 above)
Comment (imsitnikov) [reply src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:160]: Sorry, but in the latest version of the design the order should be like this:
image

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx [hunk 9]
```diff
@@ -62,115 +58,103 @@ export const WorkbookActions: React.FC<Props> = ({workbook, refreshWorkbookInfo}
...
+                                onApply: onApplyCopy,
+                                onClose: () => {
+                                    dispatch(closeDialog());
+                                },
+                            },
+                        }),
+                    );
+                },
+                text: (
+                    <>
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:121]: We prefer to use `<React.Fragment>` rather than `<>`

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx (hunk 9 above)
Comment (Sergey-weber) [reply src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx]: fixed

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx [hunk 10]
```diff
@@ -135,29 +135,29 @@ export const CollectionActions = React.memo<Props>(
 
         return (
             <div className={b(null, className)}>
-                {collectionData && (
-                    <React.Fragment>
-                        {collectionsAccessEnabled && collectionData.permissions.listAccessBindings && (
+                {collectionData && collectionData.permissions.move && (
+                    <Tooltip content={i18n('action_move')}>
+                        <Button className={b('move')} onClick={onMoveClick}>
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionActions/CollectionActions.tsx:140]: I don’t know why, but after closing the dialog tooltips appear again:
https://github.com/datalens-tech/datalens-ui/assets/29424746/61931f75-33ba-4af0-9669-204b02a856d4

The same problem does not exist on the workbook page. Need to wrap `Buttons` to `div`, and problem will be resolved:

```tsx
<Tooltip content={i18n('action_move')}>
    <div className={b('move')}>
        <Button onClick={onMoveClick}>
            <Icon data={ArrowRight} />
        </Button>
    </div>
</Tooltip>

...

<Tooltip content={i18n('action_access')}>
    <div className={b('access')}>
        <Button onClick={onEditAccessClick}>
            <Icon data={LockOpen} />
        </Button>
    </div>
</Tooltip>
```

And add `display: flex;` to:
```scss
.dl-collection-actions {
    display: flex;
    
    ...
    
}
```

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx [hunk 11]
```diff
@@ -62,109 +58,97 @@ export const WorkbookActions: React.FC<Props> = ({workbook, refreshWorkbookInfo}
...
+                            onApply: refreshWorkbookInfo,
+                            onClose: () => {
+                                dispatch(closeDialog());
+                            },
+                        },
+                    }),
+                );
+            },
+            text: (
+                <>
```
Comment (imsitnikov) [inline src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx:94]: `<>` -> `<React.Fragment>`

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx (hunk 11 above)
Comment (Sergey-weber) [reply src/ui/units/workbooks/components/WorkbookActions/WorkbookActions.tsx]: fixed

File: src/ui/units/collections/components/CollectionActions/CollectionActions.tsx (hunk 10 above)
Comment (Sergey-weber) [reply src/ui/units/collections/components/CollectionActions/CollectionActions.tsx]: fixed

**Review by imsitnikov — APPROVED** [review]

**Pull request merged**